### PR TITLE
Tdd

### PR DESCRIPTION
**Pull request by BenM4k**
In this update we created a class Solver, and implemented 3 methods inside of it (factorial, fizzbuzz and reverse). Tests were made before the class was created according to the TDD.

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  context 'when calculating factorial' do
+    it 'Should return 1' do
+      expect(solver.factorial(0)).to eq(1)
+    end
+
+    it 'should return the real answer' do
+      num = 5
+      expect(solver.factorial(num)).to eq(120)
+    end
+  end
```
Comment (TadesseDev) [inline spec/solver_spec.rb]: - Dear Team, You did great with the project, but here we need to add a test for a negative number. Remember we need to have a test for a negative value which raises an exception as an expectation. Please add this test. 🙏

**Review by TadesseDev — APPROVED** [review]
### Hi Team

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x] Most of the required tests are in place. ✅ 
 - [x]  Commit history sequence shows test are written before the methods  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AmaduKamara — APPROVED** [review]
# Status: Approved 🟢 🟢 🟢

Hi Team,

Good job working on the changes highlighted by the previous reviewer 👍🏽

Your project is complete!  There is nothing else to say other than... It's time to merge it: ship it:
Congratulations! 🎉 

## Highlights:

✅ You set up Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽
✅ Implement all methods expected for testing 👍🏽
✅ All test cases passed as expected 👍🏽
✅ README file is descriptive 👍🏽

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```
